apis/python/src/tiledbsoma/io/registration/signatures.py

```diff
@@ -0,0 +1,308 @@
+import json
+from typing import Dict, Optional
+
+import anndata as ad
+import attrs
+import pandas as pd
+import pyarrow as pa
+from typing_extensions import Self
+
+import tiledbsoma
+import tiledbsoma.logging
+from tiledbsoma._arrow_types import df_to_arrow
+
+
+def _stringify_type(t: pa.DataType) -> str:
+    """
+    Turns an Arrow data type into a stringi more suitable for logging error messages to users in a
+    distributed-computing/distributed-logging environment.
+
+    As noted in the Signature class, we pre-check logic from the ingestor.  As detailed elsewhere,
+    Arrow string and large_string must map to TileDB string, which is large-only. Thus string and
+    large_string form an equivalence class. Similarly for Arrow binary and large_binary.
+    """
+    _EQUIVALENCES = {
+        "large_string": "string",
+        "large_binary": "binary",
+    }
```

Comment:
this can either be a module-level constant or make it a regular variable name like `equivalences`